src/maze/mazeoperations.ts

```diff
@@ -3,33 +3,31 @@ import Maze from "./maze";
 import Cell from "./cell";
 
 
-// Returns an array of cells which represents the solution to the maze.
+// Returns a MazePath which represents the solution to the maze.
 export function getSolution(maze: Maze, mazePath: MazePath) {
```

Comment:
Let's refactor this function a little bit. This function doesn't return a mazepath; it just draws the solution onto the maze canvas. I suggest renaming the function to `drawSolution`.

We aren't really using the mazePath that is passed in -- all that's truly being used is the canvas object in the mazepath. We're currently forcing the caller to construct a new mazepath, which is a bit unintuitive. Let's instead create a mazepath _inside_ the function, and replace the `mazePath` parameter with a `canvas` parameter, like so:

```
export function drawSolution(maze: Maze, canvas: HTMLCanvasElement)
```

Then in `app.ts`, we can simply pass in the `this.canvas` object of the MazeGame class. We make a new MazePath to store the solution _inside_ the `drawSolution` function. This way, we can hide implementation details from client code.

```
export function drawSolution(maze: Maze, canvas: HTMLCanvasElement) {
    const mazePath = new MazePath(new MazeDrawer(maze, canvas));`
    /* Rest of the code... */
}
```

Finally, update the comment above the `drawSolution` function to say something like:

```
/**
 * Draws the solution of the given maze onto the given canvas.
 */
```